packages/react-native/React/Fabric/RCTConversions.h

```diff
@@ -52,8 +53,8 @@ inline UIColor *_Nullable RCTUIColorFromSharedColor(const facebook::react::Share
     return [UIColor whiteColor];
   }
 
-  auto components = facebook::react::colorComponentsFromColor(sharedColor);
-  return [UIColor colorWithRed:components.red green:components.green blue:components.blue alpha:components.alpha];
+  facebook::react::Color color = *sharedColor;
+  return (UIColor *)facebook::react::unwrapManagedObject(color.getUIColor());
```

Comment:
Could this be a helper method we expose in `HostPlatformColor`?

You could make it conditional on `#ifdef __OBJC__` to make it value C++.